subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,20 +859,283 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
 	return 0;
 }
 
+#ifdef CONFIG_MODEM_INFO
+
+#define MODEM_INFO_JSON_KEY_NET_INF	"networkInfo"
+#define MODEM_INFO_JSON_KEY_SIM_INF	"simInfo"
+#define MODEM_INFO_JSON_KEY_DEV_INF	"deviceInfo"
+
+static int add_modem_info_data(struct lte_param *param, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	enum at_param_type data_type;
+	int ret;
+
+	__ASSERT_NO_MSG(param != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	memset(data_name, 0, ARRAY_SIZE(data_name));
+	ret = modem_info_name_get(param->type,
```

Comment:
As of this PR, the `modem_info_name_get` is exclusively used by the cloud codec, and it is conformal with the schema laid out here:
https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/device/device.json

Given those two facts, I think it is arguable that it should be migrated to the codec as well.

@jayteemo do you agree?

Reply:
Ditto for `modem_info_type_get`

Reply:
i do not have an opinion on this, as it may increase the scope of the PR.

Reply:
`modem_info_params_get` depends on `modem_info_type_get`. For `modem_info_name_get`, you're right.

Reply:
Ah, so it does, you are right, my bad. Up to you how to proceed in light of that. Argument still stands for `modem_info_name_get`, but without also dropping `modem_info_type_get`, there is less benefit. Similar situation applies for my other comment [here](https://github.com/nrfconnect/sdk-nrf/pull/8304#discussion_r938082473)

Reply:
Let's leave this as a nice-to-have for a future PR.